### DISCUSSION TITLE
Can you "recycle" a Py5Shape?

### DISCUSSION THREAD

**Discussion by villares** [py5coding/py5generator#413] [Q&A] [answered]
Something useful for performance reasons, when drawing on off-screen buffers, like, maybe on a Py5Graphics object, is that you can create it once and point a global name to it, and then redraw over it again and again.

Is there a fast way of emptying a Py5Shape object made of vertices? I have looked at the Processing PShape methods and found only cumbersome ways of removing vertices... so I wonder if I would be able to avoid creating a new Py5Shape every time I want to draw my thousands of lines or triangles  ;)

**Comment by hx2A** [comment, marked as answer]
I have experience with recycling Py5Shapes. You don't want to create a new Py5Shape object for each frame. It is better to create one in `setup()` and re-use it.

I don't think there is a way to change the number of vertices but you can change each of the vertex values with `set_vertex()`. If you want to reduce the number of vertices you can set the vertices to values off the edge of the screen. They will be culled by the renderer (OpenGL can do this quickly).

BTW, updating a Py5Shape object's vertices with `set_vertex()` is one of the things you are allowed to do in py5's new `predraw_update()` function. It doesn't draw to the screen so you can call it whenever you wish.

**Reply by villares** [reply]
I was also thinking I can [`add_child()`](https://py5coding.org/reference/py5shape_add_child.html) and [`remove_child()`](https://py5coding.org/reference/py5shape_remove_child.html) in a Py5Shape GROUP, and maybe translate and rotate pre-made Py5Shape objects (like the triangles on my Numpy flocking sketches) I have to check if it makes things faster.

**Reply by hx2A** [reply]
Yes, definitely, all good ideas. Also note that modifying Py5Shape objects don't involve drawing to the screen, so it you like you can move that code to inside `predraw_update()`.
